server/sonar-server/src/main/java/org/sonar/server/component/ws/ProvisionedProjectsAction.java

```diff
@@ -0,0 +1,129 @@
+
+  public ProvisionedProjectsAction(DbClient dbClient) {
+    this.dbClient = dbClient;
+  }
+
+  @Override
+  public void define(WebService.NewController controller) {
+    WebService.NewAction action = controller
+      .createAction("provisioned")
+      .setDescription(
+        "Get the list of provisioned projects.<br /> " +
+          "Require admin role.")
+      .setSince("5.2")
+      .setResponseExample(Resources.getResource(getClass(), "projects-example-provisioned.json"))
+      .setHandler(this)
+      .addPagingParams(100)
+      .addFieldsParam(Arrays.asList(POSSIBLE_FIELDS));
+
+    action
+      .createParam(Param.TEXT_QUERY)
+      .setDescription("UTF-8 search query")
+      .setExampleValue("sonar");
+  }
+
+  @Override
+  public void handle(Request request, Response response) throws Exception {
+    UserSession.get().checkGlobalPermission(UserRole.ADMIN, "You need admin rights.");
+    SearchOptions options = new SearchOptions().setPage(
+      request.mandatoryParamAsInt(Param.PAGE),
+      request.mandatoryParamAsInt(Param.PAGE_SIZE)
+      );
+    List<String> desiredFields = desiredFields(request);
+    String query = request.param(Param.TEXT_QUERY);
+
+    DbSession dbSession = dbClient.openSession(false);
+    List<ComponentDto> projects;
+    int nbOfProjects;
+    try {
+      projects = dbClient.componentDao().selectProvisionedProjects(dbSession, options, query);
+      nbOfProjects = dbClient.componentDao().countProvisionedProjects(dbSession, query);
+    } finally {
+      MyBatis.closeQuietly(dbSession);
+    }
+
+    JsonWriter json = response.newJsonWriter().beginObject();
+    writeProjects(projects, json, desiredFields);
+    options.writeJson(json, nbOfProjects);
+    json.endObject().close();
+  }
+
+  private void writeProjects(List<ComponentDto> projects, JsonWriter json, List<String> desiredFields) {
+    json.name("projects");
+    json.beginArray();
+    for (ComponentDto project : projects) {
+      json.beginObject();
+      json.prop("uuid", project.uuid());
+      writeIfNeeded(json, "key", project.key(), desiredFields);
+      writeIfNeeded(json, "name", project.name(), desiredFields);
+      writeIfNeeded(json, "creationDate", project.getCreatedAt(), desiredFields);
+      json.endObject();
+    }
+    json.endArray();
+  }
+
+  private void writeIfNeeded(JsonWriter json, String fieldName, String value, List<String> desiredFields) {
+    if (desiredFields.contains(fieldName)) {
+      json.prop(fieldName, value);
+    }
+  }
+
+  private void writeIfNeeded(JsonWriter json, String fieldName, Date date, List<String> desiredFields) {
+    if (desiredFields.contains(fieldName)) {
+      json.propDate(fieldName, date);
+    }
+  }
+
+  private List<String> desiredFields(Request request) {
+    List<String> desiredFields = request.paramAsStrings(Param.FIELDS);
+    if (desiredFields == null) {
+      desiredFields = Arrays.asList(POSSIBLE_FIELDS);
+    }
+
+    return desiredFields;
```

Comment:
nope, it's not the behavior. Look at the writeProjects method and you'll see that only fields specified internally are writen in the json response.
Regarding the DB, all fields are retrieved, no partial loading. It's in the json response that there's a filter.